plan/planbuilder.go

```diff
@@ -1036,6 +1055,72 @@ func (b *planBuilder) buildExplain(explain *ast.ExplainStmt) Plan {
 	return p
 }
 
+func checkViewSelectVariable(selectstmt ast.ResultSetNode) error {
+	Fields := selectstmt.(*ast.SelectStmt).Fields.Fields
+	for _, field := range Fields {
+		ok := checkExistVariableExpr(field.Expr)
+		if ok {
+			return ErrViewSelectVariable
+		}
+	}
+	ok := checkExistVariableExpr(selectstmt.(*ast.SelectStmt).Where)
+	if ok {
+		return ErrViewSelectVariable
+	}
+	return nil
+}
+
+func checkExistVariableExpr(expr ast.ExprNode) bool {
```

Comment:
Why check this ?

Reply:
Because View's SELECT can not contains a variable or parameter

Reply:
This can be checked in validator.go